components/sup/src/manager/mod.rs

```diff
@@ -156,6 +206,26 @@ impl Manager {
         Self::new(cfg, member, fs_cfg)
     }
 
+    pub fn service_status(cfg: ManagerConfig, ident: PackageIdent) -> Result<ServiceStatus> {
+        let services = Self::status(cfg)?;
+
+        for status in services {
+            if status.package.satisfies(&ident) {
+                return Ok(status);
+            }
+        }
+
+        Err(sup_error!(Error::ServiceNotLoaded(ident)))
+    }
+
+    pub fn status(cfg: ManagerConfig) -> Result<Vec<ServiceStatus>> {
+        let state_path = Self::state_path_from(&cfg);
+        let fs_cfg = FsCfg::new(state_path);
+
+        let dat = File::open(&fs_cfg.services_data_path)?;
+        Ok(serde_json::from_reader(&dat)?)
```

Comment:
Directly converting one error to our own type is a pattern from the earlier days of our codebase but it's something that we've been trying to get away from. It's more appropriate to create a specific error and pass the error you would be converting here instead directly to it as a member. For example, you'd maybe create the error `SvcStatFileParse(serde_json::Error)` which explicitly says that there was an error in parsing the svc stat file. Since you're holding onto the `serde_json::Error` in the error struct you can display that in anyway you'd like, including just writing it into your error string at the end